### PR TITLE
Implement #else

### PR DESCRIPTION
**Pull request by hiromi-mi**
#else を実装しました. またその際必要となったいくつかの関数も追加しました.

### CODE REVIEW

**Comment by hiromi-mi** [comment]
`master` との不整合を直しました. 手元において `make selfself_test` の結果は正常でした.

**Comment by ushitora-anqou** [comment]
Could you add unit tests for `vector_peek()`, `vector_push()` and `vector_pop()` in `test.inc`?

**Comment by hiromi-mi** [comment]
Added the unit test in `test.inc` and has passed.

File: cpp.c [hunk 1]
```diff
@@ -66,11 +51,71 @@ void preprocess_tokens_detail_ifdef_ifndef(const char *keyword)
...
+    add_define(name, tokens);
+}
+
+void preprocess_tokens_detail_else()
+{
+    int *tf = vector_peek(if_stack);
+    if (tf == NULL) {
+        error("invalid #else");
+    }
+    if (*tf == 0) {
```
Comment (ushitora-anqou) [inline cpp.c]: Is that safe? `tf` may be 0 or 1, so this line may cause segmentation fault. I wonder why unit tests have passed :thinking:

Comment (ushitora-anqou) [reply cpp.c]: not L85-87, but L88

Comment (hiromi-mi) [reply cpp.c]: `tf` has `int *` type to detect `NULL`, I wonder.

Comment (ushitora-anqou) [reply cpp.c]: The value of `tf` (not `*tf`) is 0 or 1, right?

**Comment by hiromi-mi** [comment]
fixed.

**Comment by ushitora-anqou** [comment]
LGTM.

**Comment by hiromi-mi** [comment]
Looks Good To Me.

**Comment by ushitora-anqou** [comment]
Большое спасибо!

**Pull request merged**